File: PO PROJEKT 1/PO PROJEKT 1/PROJEKT PO PLIKI ŹRÓDŁOWE/Wolf.cpp
#include "Wolf.h"

Wolf::Wolf(World* world) {
	increaseStrength(9);
	setInitiative(5);
	sprite = 'w';
	this->world = world;
	spawn();
}

Wolf::Wolf(World* world, Coordinate coord) {
	increaseStrength(9);
	setInitiative(5);
	sprite = 'w';
	this->world = world;
	world->board.insert({ coord, this });
}

Organism* Wolf::createChild( Coordinate coord){
	return new Wolf(world, coord);
}

void Wolf::draw() {
	cout << GRAY_BCKG << getSprite() << RESET;
}

File: Symulator Świata/wolf.py
from animal import Animal


class Wolf(Animal):

    def __init__(self, world, coord=None):
        self.name = 'w'
        self._strength = 7
        self._initiative = 5
        self._age = 0
        self.world = world
        self.sprite = (130, 130, 150)
        if coord is not None:
            world.board[coord] = self
        else:
            self.spawn()

    def createChild(self, coord):
        wolf = Wolf(self.world, coord)
